config/config/api_platform.php

```diff
@@ -14,6 +15,7 @@
             OrderCannotBeFoundException::class => 404,
             PaymentCannotBeCancelledException::class => 400,
             UnresolvableNextCommandException::class => 400,
+            PaymentFailedException::class => 424,
```

Comment:
> The HTTP 424 Failed Dependency client error response code indicates that the method could not be performed on the resource because the requested action depended on another action, and that action failed.

It sounds perfect for me, as Tpay fails, not our app directly. However, it's not a thing I have a strong opinion about, so I can use `422` if you prefer.